File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/IosObjectiveCResources.java
package ru.pocketbyte.locolaser.platform.mobile.resource;

import ru.pocketbyte.locolaser.platform.mobile.resource.file.IosObjectiveCHResourceFile;
import ru.pocketbyte.locolaser.platform.mobile.resource.file.IosObjectiveCMResourceFile;
import ru.pocketbyte.locolaser.resource.file.ResourceFile;
import ru.pocketbyte.locolaser.summary.FileSummary;

import java.io.File;
import java.util.Set;

public class IosObjectiveCResources extends IosBaseClassResources {

    public static final String OBJC_H_FILE_EXTENSION = ".h";
    public static final String OBJC_M_FILE_EXTENSION = ".m";

    public IosObjectiveCResources(File resourcesDir, String name, String tableName) {
        super(resourcesDir, name, tableName);
    }

    @Override
    protected ResourceFile[] getResourceFiles(Set<String> locales) {
        return new ResourceFile[] {
                new IosObjectiveCHResourceFile(getObjcHFile(), getName()),
                new IosObjectiveCMResourceFile(getObjcMFile(), getName(), getTableName())
        };
    }

    @Override
    public FileSummary summaryForLocale(String locale) {
        return new FileSummary(new File[] {
                getObjcHFile(),
                getObjcMFile()
        });
    }

    private File getObjcHFile() {
        File sourceDir = getDirectory();
        sourceDir.mkdirs();
        return new File(sourceDir, getName() + OBJC_H_FILE_EXTENSION);
    }

    private File getObjcMFile() {
        File sourceDir = getDirectory();
        sourceDir.mkdirs();
        return new File(sourceDir, getName() + OBJC_M_FILE_EXTENSION);
    }
}

File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/IosObjectiveCResources.kt
package ru.pocketbyte.locolaser.platform.mobile.resource

import ru.pocketbyte.locolaser.platform.mobile.resource.file.IosObjectiveCHResourceFile
import ru.pocketbyte.locolaser.platform.mobile.resource.file.IosObjectiveCMResourceFile
import ru.pocketbyte.locolaser.resource.file.ResourceFile
import ru.pocketbyte.locolaser.summary.FileSummary

import java.io.File

class IosObjectiveCResources(
        resourcesDir: File,
        name: String,
        tableName: String?,
        filter: ((key: String) -> Boolean)?
) : IosBaseClassResources(resourcesDir, name, tableName, filter) {

    companion object {
        const val OBJC_H_FILE_EXTENSION = ".h"
        const val OBJC_M_FILE_EXTENSION = ".m"
    }

    private val objcHFile: File
        get() {
            val sourceDir = directory
            sourceDir.mkdirs()
            return File(sourceDir, name + OBJC_H_FILE_EXTENSION)
        }

    private val objcMFile: File
        get() {
            val sourceDir = directory
            sourceDir.mkdirs()
            return File(sourceDir, name + OBJC_M_FILE_EXTENSION)
        }

    override fun getResourceFiles(locales: Set<String>): Array<ResourceFile> {
        return arrayOf(IosObjectiveCHResourceFile(objcHFile, name), IosObjectiveCMResourceFile(objcMFile, name, tableName!!))
    }

    override fun summaryForLocale(locale: String): FileSummary {
        return FileSummary(arrayOf<File?>(objcHFile, objcMFile))
    }
}